multipolygon.go

```diff
@@ -51,9 +64,14 @@ func (g *MultiPolygon) Polygon(i int) *Polygon {
 	offset := 0
 	if i > 0 {
 		ends := g.endss[i-1]
-		offset = ends[len(ends)-1]
+		if len(ends) > 0 {
+			offset = ends[len(ends)-1]
+		}
 	}
 	ends := make([]int, len(g.endss[i]))
+	if len(ends) == 0 {
+		return NewPolygon(g.layout)
+	}
```

Comment:
This `if` statement can be moved to before the previous line to avoid allocating `ends` if it is empty.